state/indexer/block/kv/util.go

```diff
@@ -51,6 +51,30 @@ func heightKey(height int64) ([]byte, error) {
 	)
 }
 
+func (idx *BlockerIndexer) getEventKeys() (map[int64][][]byte, error) {
```

Comment:
My concern is this is effectively loading all the indexer keys into memory , which is potentiall 100s of GBs.